### PR TITLE
Fix: #1226: Swap wallet balances

### LINKED ISSUE #1226: Available assets should be displayed under asset selectors
**Is your feature request related to a problem? Please describe.**

It would be good to show both input and output balances in the trade selector.

**Describe the solution you'd like**

Screenshot 2024-05-30 at 11 56 56 PM

Move the wallet balance display under the input asset selector and add a new balance selector for the output asset.

### PR DESCRIPTION
**Pull request by VanishMax**
Closes #1226 

The result:

image

A trade screen now has an out-balance under the selected token. So, if users choose "Account 1 UM" as in-token and "Account 1 USDC", then the account 1's balance will be displayed. If account numbers mismatch, the "0" will be printed.

Please, take a look at the `balanceOut` memo calculation. It has a monstrous computed function and I hope you can tell how to make it neat 🙏

### CODE REVIEW

**Review by jessepinho — CHANGES_REQUESTED** [review]
Nice work! Overall looks good, but there are a couple changes I'd make to how you're getting the asset out balance. I'll mark this as "Request changes" so I can re-review, but feel free to reach out on Discord to pair on this.

**Comment by VanishMax** [comment]
Thanks, @jessepinho! Is was nice finding out about the getters. I shortened the memo function, created a couple more getters and it looks surely cleaner now

**Review by grod220 — APPROVED** [review]
_(Paired on additional commit from me)_

Good job!

**Review by jessepinho — APPROVED** [review]
LGTM! Added a couple comments, including one about the merge conflict. Merge when those are addressed!

File: apps/minifront/src/components/swap/swap-form/token-swap-input.tsx [hunk 1]
```diff
@@ -1,23 +1,57 @@
...
+  });
+};
+
+const assetOutBalanceSelector = ({ swap: { balancesResponses, assetIn, assetOut } }: AllSlices) => {
+  const match = balancesResponses.find(balance => {
+    const balanceViewMetadata = getMetadataFromBalancesResponse(balance);
+
+    return (
+      assetIn &&
+      assetOut &&
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply apps/minifront/src/components/swap/swap-form/token-swap-input.tsx]: Thanks, good addition! Changed this line

File: apps/minifront/src/components/swap/swap-form/token-swap-input.tsx [hunk 2]
```diff
@@ -74,43 +110,48 @@ export const TokenSwapInput = () => {
...
 
-            <div className='flex flex-col items-end gap-1'>
-              <AssetSelector assets={swappableAssets} value={assetOut} onChange={setAssetOut} />
-            </div>
+          <div className='flex h-full flex-col gap-2'>
+            <AssetSelector assets={swappableAssets} value={assetOut} onChange={setAssetOut} />
+            {assetOut && <BalanceValueView valueView={assetOutBalance} />}
           </div>
         </div>
-        {priceHistory.startMetadata && priceHistory.endMetadata && priceHistory.candles.length && (
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply apps/minifront/src/components/swap/swap-form/token-swap-input.tsx]: Yeah, I made a fix for this before the merge conficts. Updated it now

**Pull request merged**